### DISCUSSION TITLE
help trouble shooting on RPi 4 with Coral USB TPU

### DISCUSSION THREAD

**Discussion by jonandel** [blakeblackshear/frigate#1704] [Ask A Question] [answered]
I just received my Coral USB TPU to use with Frigate NVR running on Hassos 64 bit on RPi4 (yes the TPUs are back in stock now!) .  Up until this point Ive been running using CPU as the detectors - and its as stable as anything.

The problem Im having is configuring the detector to use the TPU makes the system very unstable, and the RPi crashes after sometimes after 10 seconds, but up to about five minutes.  Frigate add on doesnt die, but whole Pi does.  The Add in logs, and supervisor logs dont really help.  Ive tried different PSUs (no joy), and I note it took a few trial and error tries to see that the TPU can only be in certain USB ports, and can swap between the USB 3 and 2 ports - but it makes no difference.

I have SSD and TPU on USB3 and Zigbee stick in USB2 port.
I dont think its a rPi PSU problem, as this is being reported as fine.  processor and memory sensors seems to report OK values too (900MB free) and is running at around 60 degrees - which is the same irrespective of TPU running or not.
I will try a powered USB hub in next few days - but is there any more trouble shooting I can do ?

detailed Logs/debug  etc ?

My frigate.yml looks like this (attached)
[frigate.yml.txt](https://github.com/blakeblackshear/frigate/files/7109859/frigate.yml.txt)

**Comment by blakeblackshear** [comment, marked as answer]
I really need to add this to the docs. You need an externally powered USB hub. It does not matter how much power your PSU has for the Pi. It will only send so much power to the USB devices.

**Reply by jonandel** [reply]
thankx Mate - yup rock solid for about a week now !

**Reply by stamandr** [reply]
> I really need to add this to the docs. You need an externally powered USB hub. It does not matter how much power your PSU has for the Pi. It will only send so much power to the USB devices.

Do you have any recommendations on a externally powered USB hub ?
